src/implementation.cpp

```diff
@@ -16,6 +16,28 @@ namespace ada {
     return ada::parser::parse_url(input, base_url, encoding);
   }
 
+  std::string href_from_file(std::string_view input) {
+    // This is going to be much faster than constructing a URL.
+    std::string tmp_buffer;
+    std::string_view internal_input;
+    if(unicode::has_tabs_or_newline(input)) {
+      tmp_buffer = input;
+      helpers::remove_ascii_tab_or_newline(tmp_buffer);
+      internal_input = tmp_buffer;
+    } else {
+      internal_input = input;
+    }
+    std::string path;
+    if(internal_input.empty()) {
+      path = "/";
+    } else if((internal_input[0] == '/') ||(internal_input[0] == '\\')){
+      (void)helpers::parse_prepared_path(internal_input.substr(1), ada::scheme::type::FILE, path);
```

Comment:
I recommend removing `[[nodiscard]]` from the function definition instead of adding `(void)`, since this line proves that the function might run with discarding the output

Reply:
Done.